src/App.tsx

```diff
@@ -26,70 +27,129 @@ type ReorderOptions = {
   isReversed: boolean,
 };
 
-// Use this function in the render to prepare goods
+// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
   // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
+  // Sort goods if needed
+  if (sortType === SortType.ALPHABET) {
+    visibleGoods.sort(
+      (firstGood, secondGood) => firstGood.localeCompare(secondGood),
+    );
+  }
+
+  if (sortType === SortType.LENGTH) {
+    visibleGoods.sort(
+      (firstGood, secondGood) => firstGood.length - secondGood.length,
+    );
+  }
+
+  // Reverse goods if needed
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const {
+      isReversed,
+      sortType,
+    } = this.state;
+
+    const isChangesMade = isReversed || sortType !== SortType.NONE;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={
+              classNames(
+                'button is-info',
+                { 'is-light': sortType !== SortType.ALPHABET },
+              )
+            }
+            onClick={() => this.setState({
+              sortType: SortType.ALPHABET,
+            })}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={
+              classNames(
+                'button is-success',
+                { 'is-light': sortType !== SortType.LENGTH },
+              )
+            }
+            onClick={() => this.setState({
+              sortType: SortType.LENGTH,
+            })}
```

Comment:
Don't use inline callbacks, we don't need to create a new arrow function for each re-render

Reply:
Create class methods